test/WebSites/BasicWebSite/Controllers/ContentNegotiation/NormalController.cs

```diff
@@ -1,28 +1,70 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Buffers;
 using BasicWebSite.Formatters;
 using BasicWebSite.Models;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Filters;
 using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.Extensions.Options;
 using Newtonsoft.Json;
 
 namespace BasicWebSite.Controllers.ContentNegotiation
 {
     public class NormalController : Controller
     {
+        private JsonOutputFormatter _indentingFormatter;
+
+        public NormalController(IOptions<MvcJsonOptions> jsonOptions, ArrayPool<char> charPool)
+        {
+            var defaultSettings = jsonOptions.Value.SerializerSettings;
+            var settings = new JsonSerializerSettings
```

Comment:
> we should provide an easier way for end users to get these settings.

These settings come from very few places: `SerializerSettingsProvider.CreateSerializerSettings()` and the user's own code in `Startup` or a resource or action filter e.g. an `OnActionExecuting()` override in their controller. If needed, we could move `SerializerSettingsProvider` into a non-`.Internal` namespace.